src/tests/ftest/util/pool_test_base.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from itertools import product
+
+from avocado.core.exceptions import TestFail
+
+from apricot import TestWithServers
+from command_utils import BasicParameter
+from test_utils import TestPool
+
+
+class PoolTestBase(TestWithServers):
+    """Defines common methods for pool tests.
+
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up each test case."""
+        # Generate separate log files per test variant
+        self.update_log_file_names()
+        super(PoolTestBase, self).setUp()
+
+    def get_param_list(self, name):
+        """Get the list of param values and expected result for the param name.
+
+        Args:
+            name (str): parameter name
+
+        Returns:
+            list: a list of lists containing the parameter name, value, and a
+                boolean indicating whether or not this value will succeed in
+                creating a pool.
+
+        """
+        # Get the default value for the parameter to use if a list is not set
+        default_value = getattr(self.pool, name)
+        if isinstance(default_value, BasicParameter):
+            default_value = default_value.value
+
+        # Get the list of parameter values and pass/fail state from the yaml
+        param_list = self.params.get(
+            "{}_list".format(name),
+            self.pool.namespace,
+            [[default_value, "PASS"]]
+        )
+
+        # Process the list data
+        for index in range(len(param_list)):
```

Comment:
(pylint-consider-using-enumerate) Consider using enumerate instead of iterating with range and len